File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/mediator/MainMediator.java
package designPatterns.behavioral.mediator;

public class MainMediator {
  public static void main(String[] args) {
    Mediator mediator = new ConcreteMediator();
    Model m1 = new ConcreteModel(mediator, "val1");
    Model m2 = new ConcreteModel(mediator, "val2");
    mediator.addModel(m1);
    mediator.addModel(m2);

    System.out.println(m1.send("Hello everyone!"));
    System.out.println(m2.receive("Hello"));
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/mediator/MainMediator.kt
package behavioral.mediator

fun main() {
  var mediator = ConcreteMediator()
  var m1 = ConcreteModel(mediator, "val1")
  var m2 = ConcreteModel(mediator, "val2")
  mediator.addModel(m1)
  mediator.addModel(m2)

  println(m1.send("Hello everyone!"))
  println(m2.receive("Hello"))
}